src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -1180,6 +1180,13 @@ public RequestLineStatus TakeStartLine(SocketInput input)
 
                 return RequestLineStatus.Done;
             }
+            catch (DecodingException)
+            {
+                RejectRequest(RequestRejectionReason.NonAsciiOrNullCharactersInRequestLine,
+                    Log.IsEnabled(LogLevel.Information) ? start.GetAsciiStringEscaped(end, MaxInvalidRequestLineChars) : string.Empty);
+                // won't run but compiler complains
```

Comment:
Fine either way, but for readability I actually prefer the explicit throw. The first time I read this code, it was not immediately obvious that 'RejectRequest' and its overloads terminate execution.